### ISSUE TITLE
Try does not catch undefined variable

### ISSUE THREAD

**Issue by capital-G** [supercollider/supercollider#5921]
## Environment

* SuperCollider version: `3.11.2`
* Operating system: macOS 12.6.1
* Other details (Qt version, audio driver, etc.): -

## Steps to reproduce

```supercollider
try {abc} {|e| "error".postln;}
```

returns

```
ERROR: Variable 'abc' not defined.
  in interpreted text
  line 2 char 8:

  try {abc} {|e| e.postln;}
          
  ) 
```

instead of `error`.

## Expected vs. actual behavior

I would thin that `try` allows me to catch any exception.

E.g. python allows to do such things

```python
In [1]: try:
   ...:     print(xcd)
   ...: except NameError as e:
   ...:     print(e)
   ...:
name 'xcd' is not defined
```

This comes in handy if you want to execute arbitrary code via `String.interpret` and want to see if it succeeded or catch if something did not work.

**Comment by dyfer** [comment]
I've came across this as well, but...

https://github.com/supercollider/supercollider/issues/1514#issuecomment-258550163

IIUC `try` would only catch compilation errors, but undefined variable is a syntax check failure. I agree it would be nice to be able to catch syntax check errors too, but I'm not sure if that's feasible in sclang...

**Comment by telephon** [comment]
yes, this would be useful for a number of things. Couldn't the interpreter call back sclang and throw an error there? Currently, it seems, there is only one way back. The (possibly silly) hack would be for the interpreter to replace the code by something like "SyntaxError(%).throw", with % replaced by the post, and interpret this, instead of just posting the info.

**Comment by JordanHendersonMusic** [comment]
Exceptions are implemented in the language. In sc, you must know the frame index of the variable to be able to compile the code ­— variable (with the exception of environment variables) are resolved at compile time. Meaning you just can't throw an exception without having valid code.

We could have another method that takes in the code as a string and attempts to see if it compiles. The methods we have to do this right now all return `nil` if it fails which is really annoying for testing purposes.

**Comment by JordanHendersonMusic** [comment]
I've rephrased this issue with a plan for how to do this #7609
